src/sync/sync-api.js

```diff
@@ -125,34 +131,94 @@ export class SyncApi extends TypedEmitter {
    */
   #getState(namespaceSyncState) {
     const state = reduceSyncState(namespaceSyncState)
-    state.data.syncing = this.#isSyncing
+
+    switch (this.#previousSyncEnabledState) {
+      case 'none':
+        state.initial.isSyncEnabled = state.data.isSyncEnabled = false
+        break
+      case 'presync':
+        state.initial.isSyncEnabled = true
+        state.data.isSyncEnabled = false
+        break
+      case 'all':
+        state.initial.isSyncEnabled = state.data.isSyncEnabled = true
+        break
+      default:
+        throw new ExhaustivenessError(this.#previousSyncEnabledState)
+    }
+
     return state
   }
 
+  #updateState = () => {
```

Comment:
Maybe add namespace sync state as an optional parameter, to save recalculating it on state events. It's expensive to calculate.

Reply:
Done in db4ff6c6318bc43897c5bf84ff8c60ece7e6ef67.